bentoml/_internal/types.py

```diff
@@ -48,27 +48,26 @@
 JSON_CHARSET = "utf-8"
 
 if TYPE_CHECKING:
-    PathType = t.Union[str, os.PathLike[str]]
+    PathType = str | os.PathLike[str]
 else:
     PathType = t.Union[str, os.PathLike]
 
-MetadataType: t.TypeAlias = t.Union[
-    str,
-    bytes,
-    bool,
-    int,
-    float,
-    complex,
-    datetime,
-    date,
-    time,
-    timedelta,
-    t.List["MetadataType"],
-    t.Tuple["MetadataType"],
-    t.Dict[str, "MetadataType"],
+ValueType = t.Union[
+    str, bytes, bool, int, float, complex, datetime, date, time, timedelta
+]
+MetadataType = t.Union[
+    ValueType,
+    t.List[ValueType],
+    t.Dict[str, ValueType],
+    t.Dict[t.Union[str, float, int], ValueType],
+    t.Tuple[ValueType],
 ]
```

Comment:
Is this change necessary? it's not exactly correct.

 I think only the l66-70 is what's required for cattrs to work.

Reply:
oh no, let me revert this back. 

You are correct. Only L66-70 is needed.